src/build/mod.rs

```diff
@@ -225,33 +226,34 @@ async fn process_dockerfile<R: AsyncRead + std::marker::Unpin>(
 
     let repro_time = r#"find $( ls / | grep -E -v "^(dev|mnt|proc|sys)$" ) -xdev | xargs touch --date="@0" --no-dereference || true"#.to_string();
 
-    let mut env_directives = vec![
-        // set cage name and app uuid as in enclave env vars
-        Directive::new_env("EV_CAGE_NAME", build_config.cage_name()),
-        Directive::new_env("CAGE_UUID", build_config.cage_uuid()),
-        Directive::new_env("EV_APP_UUID", build_config.app_uuid()),
-        Directive::new_env("EV_TEAM_UUID", build_config.team_uuid()),
-        Directive::new_env("DATA_PLANE_HEALTH_CHECKS", "true"),
-        Directive::new_env("EV_API_KEY_AUTH", &build_config.api_key_auth().to_string()),
-        Directive::new_env(
-            "EV_TRX_LOGGING_ENABLED",
-            &build_config.trx_logging_enabled().to_string(),
-        ),
-    ];
-
     let egress = build_config.clone().egress;
-    if egress.is_enabled() {
-        let ports = Directive::new_env("EGRESS_PORTS", &egress.clone().get_ports());
-        env_directives.push(ports);
-        let domains = Directive::new_env("EV_EGRESS_ALLOW_LIST", &egress.get_destinations());
-        env_directives.push(domains)
+    let egress_settings = if egress.is_enabled() {
+        json!({
+            "ports": &egress.clone().get_ports(),
+            "allowList": &egress.get_destinations()
+        })
+    } else {
+        json!({})
     };
 
+    let dataplane_info = json!({
+        "apiKeyAuth":  &build_config.api_key_auth().to_string(),
+        "trxLoggingEnabled": &build_config.trx_logging_enabled().to_string(),
+        "egress": egress_settings
+    })
+    .to_string();
+
+    let dataplane_env = format!(
+        "echo {} > /etc/dataplane-vars",
```

Comment:
Is there a reason why this doesn't have an extension given that it's json? Also would be nice to have a more obvious name, something like "dataplane-config"?

Reply:
good point, just updated it there!